scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,168 @@
+import numpy as np
+from functools import partial
+from scipy import stats
+
+
+def _bws_input_validation(x, y, alternative, method):
+    ''' Input validation and standardization for bws test'''
+    x, y = np.atleast_1d(x, y)
+    if x.ndim > 1 or y.ndim > 1:
+        raise ValueError('`x` and `y` must be exactly one-dimensional.')
+    if np.isnan(x).any() or np.isnan(y).any():
+        raise ValueError('`x` and `y` must not contain NaNs.')
+    if np.size(x) == 0 or np.size(y) == 0:
+        raise ValueError('`x` and `y` must be of nonzero size.')
+
+    z = stats.rankdata(np.concatenate((x, y)))
+    x, y = z[:len(x)], z[len(x):]
+
+    alternatives = {'two-sided', 'less', 'greater'}
+    alternative = alternative.lower()
+    if alternative not in alternatives:
+        raise ValueError(f'`alternative` must be one of {alternatives}.')
+
+    method = stats.PermutationMethod() if method is None else method
+    if not isinstance(method, stats.PermutationMethod):
+        raise ValueError('`method` must be an instance of '
+                         '`scipy.stats.PermutationMethod`')
+
+    return x, y, alternative, method
+
+
+def _bws_statistic(x, y, alternative):
+    '''Compute the BWS test statistic for two independent samples'''
+    Ri, Hj = np.sort(x, axis=-1), np.sort(y, axis=-1)
+    n, m = Ri.shape[-1], Hj.shape[-1]
+    i, j = np.arange(1, n+1), np.arange(1, m+1)
+
+    Bx_num = (Ri - (m + n)/n * i)
+    Bx_num *= Bx_num if alternative == 'two-sided' else np.abs(Bx_num)
```

Comment:
I find this line hard to read, could this be re-written as 
```
if ...:
   ...
else:
   ...
```

Reply:
Refactored for readability.